timelock-agent/src/main/java/com/palantir/timelock/clock/ClockCreator.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.clock;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import javax.net.ssl.SSLSocketFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.atlasdb.timelock.clock.ClockServiceImpl;
+import com.palantir.atlasdb.timelock.clock.ClockSkewMonitor;
+import com.palantir.remoting2.config.ssl.SslSocketFactories;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.paxos.PaxosRemotingUtils;
+
+public class ClockCreator {
```

Comment:
Sure, will change this. I think this also creates a `ClockServiceImpl`, though that's a part of the skew-monitor in any case.